cmake/CeleritasUtils.cmake

```diff
@@ -154,6 +154,20 @@ function(celeritas_generate_empty_cu_file emptyfilenamevar target)
   set(${emptyfilenamevar} ${_stub} PARENT_SCOPE)
 endfunction()
 
+#
+# Transfer the setting \${what} (both the PUBLIC and INTERFACE version) to from library \${fromlib} to the library \${tolib} that depends on it
+#
+function(celeritas_transfer_setting fromlib tolib what)
+  get_target_property(_temp ${fromlib} ${what})
+  if (_temp)
+    cmake_language(CALL target_${what} ${tolib} PUBLIC ${_temp})
```

Comment:
As you said, this feature requires CMake 3.18 and higher. Can you update the vecgeom+cuda version restriction in the main CMakeLists.txt to reflect the new minimum? Right now it has 3.17 as the minimum (and a comment that's not applicable since the new `celeritas_library` functions you wrote).

Reply:
I pushed the requested commit.  Note that the use of `cmake_language` is solely to reduce code duplication (i.e. it is not "necessary").